minet/bluesky/http_client.py

```diff
@@ -101,30 +110,136 @@ def request(
             headers=headers,
         )
 
+        if response.status < 200 or response.status >= 300:
```

Comment:
I don't expect the status to go in the `1xx` range. This is usually unused. Identically, you should test for `>= 400` because `3xx` are redirections, are handled by `request` and will not happen here and would not be considered errors.